subsys/bluetooth/services/bas_c.c

```diff
@@ -113,14 +113,124 @@ static u8_t read_process(struct bt_conn *conn, u8_t err,
 	}
 
 	bas_c->read_cb = NULL;
+
 	return BT_GATT_ITER_STOP;
 }
 
+/**
+ * @brief Process periodic battery level value read
+ *
+ * Internal function to process report read and pass it further.
+ * And the end the new read request is triggered.
+ *
+ * @param conn   Connection handler.
+ * @param err    Read ATT error code.
+ * @param params Notification parameters structure - the pointer
+ *               to the structure provided to read function.
+ * @param data   Pointer to the data buffer.
+ * @param length The size of the received data.
+ *
+ * @retval BT_GATT_ITER_STOP     Stop notification
+ * @retval BT_GATT_ITER_CONTINUE Continue notification
+ */
+static u8_t periodic_read_process(struct bt_conn *conn, u8_t err,
+				  struct bt_gatt_read_params *params,
+				   const void *data, u16_t length)
+{
+	struct bt_gatt_bas_c *bas_c;
+	u8_t battery_level = BT_GATT_BAS_VAL_INVALID;
+	const u8_t *bdata = data;
+
+	bas_c = CONTAINER_OF(params, struct bt_gatt_bas_c,
+			periodic_read.params);
+
+	if (!bas_c->notify_cb) {
+		LOG_ERR("No notification callback present");
+	} else  if (err) {
+		LOG_ERR("Read value error: %d", err);
+	} else if (!data || length != 1) {
+		LOG_ERR("Unexpected read value size.");
```

Comment:
Shouldn't give error on something that can be the fault of the remote, I'd switch to warning here.

Reply:
Why not? We should not assert it, but I do not see a good reason not to log an error.

Reply:
It's just semantics. 
Use LOG__ERR for errors that are unrecoverable.
Use LOG_WRN for errors that are recoverable.

Logging an unrecoverable error on something that is outside of our control is bad in my opinion.

Anyway. I'll leave it up to you to decide which log level to use.